src/dc_jobthread.rs

```diff
@@ -7,23 +7,24 @@ use crate::dc_sqlite3::*;
 use crate::dc_tools::*;
 use crate::imap::Imap;
 use crate::x::*;
+use std::ffi::CString;
 
 #[repr(C)]
 pub struct dc_jobthread_t {
-    pub name: *mut libc::c_char,
-    pub folder_config_name: *mut libc::c_char,
+    pub name: &'static str,
```

Comment:
this should be `String`, not a static str

Reply:
I thought about this for a long time and here is why I decided for `&'static str`: 
https://doc.rust-lang.org/1.7.0/book/strings.html:

> 
> let greeting = "Hello there."; // greeting: &'static str
> 
> "Hello there." is a string literal and its type is &'static str. A string literal is a string slice that is statically allocated, meaning that it’s saved inside our compiled program, and exists for the entire duration it runs. The greeting binding is a reference to this statically allocated string. Any function expecting a string slice will also accept a string literal.

And as `name` is always assigned a string literal that has to be there in the binary anyway (and already is 'static str), it is better to do it this way than cloning it to get a String and then have the string in the RAM twice. 

Did I persuade you?

Reply:
(Once the string literal is "SENTBOX\x00" and once "MVBOX\x00")

Reply:
good catch